File: frontend/lib/model/trainer.dart
import 'package:poligon/model/path.dart';

/// Author: Łukasz Piętka (FUT 2025)
class Trainer {
  final int id;
  final String name;
  final TrainingPath path;
  final String description;
  final String imageUrl;
  final String trainings;
  final DateTime? lastModified;
  double imageAlignmentY;

  Trainer({
    required this.id,
    required this.name,
    required this.path,
    required this.description,
    required this.trainings,
    required this.imageUrl,
    this.lastModified,
    required this.imageAlignmentY,
  });
}

File: backend/src/main/java/me/havooc/poligon/model/Trainer.java
/**
 * @Author Łukasz Piętka (FUT 2025)
 */
package me.havooc.poligon.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Trainer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    @Enumerated(EnumType.STRING)
    private TrainingPath path;
    @Column(length = 4096)
    private String description;
    private String imageUrl;
    @Column(length = 1028)
    private String trainings;
    @UpdateTimestamp
    private LocalDateTime lastModified;
    private Double imageAlignmentY;
}